src/components/Header/Header.tsx

```diff
@@ -0,0 +1,110 @@
+import classNames from 'classnames';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { TodosContext } from '../TododsContext/TodosContext';
+import { addTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setTempTodo: (todo: Todo | null) => void,
+};
+
+export const Header: React.FC<Props> = ({ setTempTodo }) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const {
+    todos, setTodos, setErrorMessage, USER_ID,
+  } = useContext(TodosContext);
+  const titleRef = useRef<HTMLInputElement | null>(null);
+  const [isSubmiting, setIsSubmiting] = useState(false);
+
+  useEffect(() => {
+    if (titleRef.current) {
+      titleRef.current.focus();
+    }
+  });
+
+  const handleChangeTodoTitle = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (todoTitle.trim() === '') {
+      setErrorMessage('Title should not be empty');
+    } else {
+      setIsSubmiting(true);
+
+      setTempTodo({
+        title: todoTitle,
+        completed: false,
+        userId: USER_ID,
+        id: 0,
+      });
+
+      addTodo({
+        title: todoTitle,
+        completed: false,
+        userId: USER_ID,
+      })
+        .then((newTodo: Todo) => {
+          setTodos(currentTodos => [...currentTodos, newTodo]);
+          setTodoTitle('');
+        })
+        .catch((error) => {
+          setErrorMessage('Unable to add a todo');
+          throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)